kernel/include/kswap.h

```diff
@@ -126,12 +122,6 @@ static inline int z_swap_irqlock(unsigned int key)
 #endif
 #endif
 	ret = __swap(key);
-#ifndef CONFIG_ARM
```

Comment:
how would architectures other than ARM and ARC call this function now? You are only changing this for ARC in the arch code.

Reply:
`sys_trace_thread_switched_in` should not be in `kswap.h`. when return form `__swap`, it will back to previous thread.
Other architectures also need to do the same fix like ARC to let tracing function work.

Reply:
@IRISZZW Do you mean that the test is broken for other architectures anyway, so removing the #ifndef doesn't matter anyway?
Given that this is a stable branch, I'd say let's not run the risk of changing more than required, and keep the #ifndef but extend with ARC.

Reply:
OK, I will keep it.